### PR TITLE
zh-cn: init translation of svelte deployment

### PR DESCRIPTION
**Pull request by hanyujie2002**
### Related issues and pull requests

Relates to #16965

### CODE REVIEW

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_deployment_next/index.md [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+我们将把我们的待办事项应用部署到 [GitLab Pages](https://about.gitlab.com/stages-devops-lifecycle/pages/)，借此来演示这一点。
+
+1. 首先，你需要[在 GitLab 注册](https://gitlab.com/users/sign_up)，然后[创建一个新项目](https://gitlab.com/projects/new)。给你的新项目取一个简短、简单的名称，比如“mdn-svelte-todo”。你将获得指向你的新 GitLab 存储库的远程 URL，例如 `git@gitlab.com:[你的用户名]/[你的项目名].git`。
+2. 在开始上传内容到你的 git 存储库之前，最好的做法是添加 `.gitignore` 文件，告诉 git 哪些文件应该排除在源代码控制之外。在我们的例子中，我们将在你本地项目的根文件夹中创建 `.gitignore` 文件，内容如下：
+
+   ```bash
+   node_modules/
+   ```
+
+3. 现在让我们回到 GitLab。在创建新仓库后，GitLab 将向你打招呼，并解释了上传现有文件的不同选项。按照 _Push an existing folder_ （推送现有文件夹）标题下的步骤进行操作：
```
Comment (hanyujie2002) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_deployment_next/index.md:211]:
```suggestion
3. 现在让我们回到 GitLab。在创建新仓库后，GitLab 将向你打招呼，并解释了上传现有文件的不同选项。按照 *Push an existing folder*（推送现有文件夹）标题下的步骤进行操作：
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_deployment_next/index.md [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+     artifacts:
+       paths:
+         - public
+     only:
+       - main
+   ```
+
+在这里，我们告诉 GitLab 使用最新版本的 Node 镜像来构建我们的应用程序。接下来，我们声明了一个 `pages` 作业，以启用 GitLab Pages。每当我们对仓库进行推送时，GitLab 将运行 `npm install` 和 `npm run build` 来构建我们的应用程序。我们还告诉 GitLab 部署 `public` 文件夹的内容。在最后一行，我们配置 GitLab 仅在对主分支进行推送时重新部署我们的应用程序。
+
+2. 由于我们的应用程序将在子目录发布（例如 `https://your-user.gitlab.io/mdn-svelte-todo`），我们需要使 `public/index.html` 文件中对 JavaScript 和 CSS 文件的引用变为相对路径。为了做到这一点，我们只需从 `/global.css`、`/build/bundle.css` 和 `/build/bundle.js` 的 URL 中去掉前面的斜杠(`/`)，像这样：
```
Comment (hanyujie2002) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_deployment_next/index.md:248]:
```suggestion
2. 由于我们的应用程序将在子目录发布（例如 `https://your-user.gitlab.io/mdn-svelte-todo`），我们需要使 `public/index.html` 文件中对 JavaScript 和 CSS 文件的引用变为相对路径。为了做到这一点，我们只需从 `/global.css`、`/build/bundle.css` 和 `/build/bundle.js` 的 URL 中去掉前面的斜杠（`/`），像这样：
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_deployment_next/index.md [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+       - npm install
+       - npm run build
+     artifacts:
+       paths:
+         - public
+     only:
+       - main
+   ```
+
+在这里，我们告诉 GitLab 使用最新版本的 Node 镜像来构建我们的应用程序。接下来，我们声明了一个 `pages` 作业，以启用 GitLab Pages。每当我们对仓库进行推送时，GitLab 将运行 `npm install` 和 `npm run build` 来构建我们的应用程序。我们还告诉 GitLab 部署 `public` 文件夹的内容。在最后一行，我们配置 GitLab 仅在对主分支进行推送时重新部署我们的应用程序。
```
Comment (hanyujie2002) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_deployment_next/index.md:246]:
```suggestion
   在这里，我们告诉 GitLab 使用最新版本的 Node 镜像来构建我们的应用程序。接下来，我们声明了一个 `pages` 作业，以启用 GitLab Pages。每当我们对仓库进行推送时，GitLab 将运行 `npm install` 和 `npm run build` 来构建我们的应用程序。我们还告诉 GitLab 部署 `public` 文件夹的内容。在最后一行，我们配置 GitLab 仅在对主分支进行推送时重新部署我们的应用程序。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_deployment_next/index.md [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+   ```bash
+   cd your_root_directory # 进入项目的根文件夹
+   git init
+   git remote add origin https://gitlab.com/[你的用户名]/mdn-svelte-todo.git
+   git add .
+   git commit -m "Initial commit"
+   git push -u origin main
+   ```
+
+   > **备注：** 你可以使用 [git 协议](https://git-scm.com/book/zh/v2/%E6%9C%8D%E5%8A%A1%E5%99%A8%E4%B8%8A%E7%9A%84Git-%E5%8D%8F%E8%AE%AE#_the_git_protocol)替代 `https`，它更快速，而且无需每次访问远程仓库时输入用户名和密码。要使用 git 协议，你需要[创建一个 SSH 密钥对](https://docs.gitlab.com/ee/user/ssh.html#generate-an-ssh-key-pair)。你的远程仓库 URL 将是这样的：`git@gitlab.com:[你的用户名]/mdn-svelte-todo.git`。
```
Comment (JasonLamv-t) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_deployment_next/index.md:222]:
```suggestion
   > **备注：** 你可以使用 [git 协议](https://git-scm.com/book/zh/v2/服务器上的-Git-协议#_git_协议)替代 `https`，它更快速，而且无需每次访问远程仓库时输入用户名和密码。要使用 git 协议，你需要[创建一个 SSH 密钥对](https://docs.gitlab.cn/jh/user/ssh.html#生成-ssh-密钥对)。你的远程仓库 URL 将是这样的：`git@gitlab.com:[你的用户名]/mdn-svelte-todo.git`。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_deployment_next/index.md [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+
+在这个部分，我们将为你提供一些资源和项目，以进一步学习 Svelte。
+
+### Svelte 文档
+
+要深入了解 Svelte 并学习更多关于它的知识，你一定要访问 [Svelte 主页](https://svelte.dev/)。在那里，有[许多文章](https://svelte.dev/blog)解释 Svelte 的理念。如果你还没有这样做，一定要完成 [Svelte 交互式教程](https://learn.svelte.dev/)。我们已经涵盖了大部分内容，所以你不需要花费太多时间来完成它，你可以将其视为实践！
+
+你还可以查阅 [Svelte API 文档](https://svelte.dev/docs)和可用的[示例代码](https://svelte.dev/examples/hello-world)。
+
+要了解 Svelte 作者创建它背后的动力，你应该观看 [Rich Harris](https://twitter.com/Rich_Harris) 在 YouTube 上的 [Rethinking reactivity](https://www.youtube.com/watch?v=AdNJ3fydeao&t=47s)（重新思考响应式）演讲。他是 Svelte 的创作者，所以他对此有很多见解。你还可以在这里找到交互式幻灯片，这些幻灯片显然是使用 Svelte 构建的。如果你喜欢这个演讲，你还可以观看 Rich Harris 在 [JSCAMP 2019](https://jscamp.tech/2019/) 上的演讲 [The Return of 'Write Less, Do More'](https://www.youtube.com/watch?v=BzX4aTRPzno)（“写的少，做的多”的回归）。
```
Comment (JasonLamv-t) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_deployment_next/index.md:293]:
```suggestion
要了解 Svelte 作者创建它背后的动机，你应该观看 [Rich Harris](https://twitter.com/Rich_Harris) 在 YouTube 上的 [Rethinking reactivity（重新思考响应式）](https://www.youtube.com/watch?v=AdNJ3fydeao&t=47s)演讲。他是 Svelte 的创作者，所以他对此有很多见解。你还可以在这里找到交互式幻灯片，当然，这些幻灯片是使用 Svelte 构建的。如果你喜欢这个演讲，你还可以观看 Rich Harris 在 [JSCAMP 2019](https://jscamp.tech/2019/) 上的演讲 [The Return of 'Write Less, Do More'（“写的少，做的多”的回归）](https://www.youtube.com/watch?v=BzX4aTRPzno)。
```

**Review by JasonLamv-t — APPROVED** [review]
Thank you! @hanyujie2002

**Pull request merged**